### PR TITLE
Chore/extend pinning tests

### PR DESCRIPTION
**Pull request by whiskey**
## Description
Crash reports pointing to `SecTrustEvaluateAsyncWithError` in `CoronaWarnURLSessionDelegate` lead to some extended checks and tests.

## Link to Jira
[EXPOSUREAPP-7595](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7595)
This issue might not be related to any pinning issues but triggered this PR

## Screenshots
(enter typical 'hacker wearing a black hoodie'-image here)

### CODE REVIEW

**Review by SwissMountainFrog — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Client/Security/__tests__/HTTPClientCertificatePinningTests.swift [hunk 1]
```diff
@@ -51,4 +52,50 @@ class HTTPClientCertificatePinningTests: CWATestCase {
...
+		let environments = [EnvironmentDescriptor.production, EnvironmentDescriptor.custom("wru")]
+		environments.forEach { descriptor in
+			let env = Environments().environment(descriptor)
+			let hosts = [
+				env.dataDonationURL,
+				env.distributionURL,
+				env.errorLogSubmissionURL,
+				env.submissionURL,
+				env.verificationURL
+				// TODO: add certificate host URL //swiftlint:disable:this todo
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Client/Security/__tests__/HTTPClientCertificatePinningTests.swift:66]: ToDo with swiftlint disable, why not use a simple comment here?

**Review by ArturFriesen — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Client/Security/__tests__/HTTPClientCertificatePinningTests.swift (hunk 1 above)
Comment (whiskey) [reply src/xcode/ENA/ENA/Source/Client/Security/__tests__/HTTPClientCertificatePinningTests.swift:66]: I'll check for TODOs locally. In the past someone came up with splitting them to "TO DO" with the result of missing out tons of stuff and producing issues all over the place.

Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Client/Security/__tests__/HTTPClientCertificatePinningTests.swift:66]: What about calling it "TODOdubidubidu"? 🤪

Comment (whiskey) [reply src/xcode/ENA/ENA/Source/Client/Security/__tests__/HTTPClientCertificatePinningTests.swift]: I'll leave it in as long as I struggle to get [Kai's PR](https://github.com/corona-warn-app/cwa-app-ios/pull/2835) through our beloved CI.

**Review by nickguendling — APPROVED** [review]

**Pull request merged**